default.nix

```diff
@@ -43,18 +43,35 @@ rec {
       inherit name;
       path = root;
       filter = path: type:
-        (builtins.any (f: f path type) include_) &&
-        (!builtins.any (f: f path type) exclude_);
+        (builtins.any (f: f root path type) include_) &&
+        (!builtins.any (f: f root path type) exclude_);
     };
 
+  # Match a directory and any path inside of it
+  inDirectory = directory:
+    root: path: type:
+      let
+        # Convert `directory` to a path to clean user input
+        dirAsPath =
+          if builtins.isString directory then root + ("/" + directory)
+          else if builtins.isPath directory then directory
+          else
+            throw "inDirectory: Unsupported type ${builtins.typeOf directory}, expected string or path";
```

Comment:
Not sure if the conversion could be pushed up to the core filter since the path string is an argument to `inDirectory`. Though we could probably leverage partial application to the same effect. Would something like this work?

In `inDirectory`:

```nix
inDirectory = directory:
  let
    dirAsPath =
      if builtins.isString directory then root + ("/" + directory)
      else if builtins.isPath directory then directory
      else
        throw "inDirectory: Unsupported type ${builtins.typeOf directory}, expected string or path";
  in
  root: path: type:
    let
      directory_ = toString dirAsPath;
      ...
```

In `toMatcher`:

```nix
if builtins.isFunction f then f root
```

Reply:
Good point.

Reply:
I pushed the proposed changes as I had them in mind. Let me know what you think!

Reply:
Looks mostly good. I don't have time to test much until later, but I caught one error which I commented on.

Good idea moving the path cleaning to a helper function.